x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.test.ts

```diff
@@ -1252,7 +1258,97 @@ describe('IndexPattern Data Source', () => {
           layerId: 'first',
         });
 
-        expect(publicAPI.getTableSpec()).toEqual([{ columnId: 'col2' }]);
+        expect(publicAPI.getTableSpec()).toEqual([expect.objectContaining({ columnId: 'col2' })]);
+      });
+
+      it('should collect all fields (also from referenced columns)', () => {
+        publicAPI = indexPatternDatasource.getPublicAPI({
+          state: {
+            ...enrichBaseState(baseState),
+            layers: {
+              first: {
+                indexPatternId: '1',
+                columnOrder: ['col1', 'col2'],
+                columns: {
+                  col1: {
+                    label: 'Sum',
+                    dataType: 'number',
+                    isBucketed: false,
+
+                    operationType: 'sum',
+                    sourceField: 'test',
+                    params: {},
+                  } as GenericIndexPatternColumn,
+                  col2: {
+                    label: 'Cumulative sum',
+                    dataType: 'number',
+                    isBucketed: false,
+
+                    operationType: 'cumulative_sum',
+                    references: ['col1'],
+                    params: {},
+                  } as GenericIndexPatternColumn,
+                },
+              },
+            },
+          },
+          layerId: 'first',
+        });
+
+        expect(publicAPI.getTableSpec()).toEqual([{ columnId: 'col2', fields: ['test'] }]);
```

Comment:
I understand what is going on here, but could you explain why we only return the column ID for root of the reference chain?